### PR TITLE
[CI] UT logs post op and artifact

### PR DESCRIPTION
**Pull request by RUIJIEZHONG66166**

### CODE REVIEW

File: .github/workflows/_linux_ut.yml [hunk 1]
```diff
@@ -28,6 +28,12 @@ on:
         type: string
         default: 'linux.idc.xpu'
         description: Runner label
+      expected_fail_num:
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:31]: Do we really need this parameter?

File: .github/workflows/_linux_ut.yml [hunk 2]
```diff
@@ -141,3 +148,81 @@ jobs:
...
+      - name: Upload Inductor XPU UT Data
+        uses: actions/upload-artifact@v4
+        with:
+          name: Inductor-XPU-UT-Data-${{ github.event.pull_request.number || github.sha }}
+          path: ${{ github.workspace }}/ut_log
+
+      - name: UT Test Results Check
+        if: ${{ inputs.expected_fail_num }}
+        shell: bash
+        run: |
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:161]: Can we abstract this step as a script and put it into `.github/scripts`? and in this step, we not only check the test pass or not, but also show the failure cases and logs directly

File: .github/workflows/_linux_ut.yml [hunk 3]
```diff
@@ -142,3 +144,31 @@ jobs:
             test_cmd="${test_cmd} test_xpu.py"
           fi
           eval $test_cmd
+
+      - name: Upload Inductor XPU UT Data
+        uses: actions/upload-artifact@v4
+        with:
+          name: Inductor-XPU-UT-Data-${{ github.event.pull_request.number || github.sha }}
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:151]: Change the name to `Torch-XPU-UT-Log-xxx`. And we can move the upload step after the result check step with `if: always()`

File: .github/workflows/_linux_ut.yml [hunk 4]
```diff
@@ -120,7 +122,7 @@ jobs:
             if [[ "$xpu_case" != *"*"* && "$xpu_case" != *.so && "$xpu_case" != *.a ]]; then
               case_name=$(basename "$xpu_case")
               echo "Testing ${case_name} ..."
-              "$xpu_case" --gtest_output=xml:"$TEST_REPORTS_DIR"/"$case_name".xml 
+              "$xpu_case" --gtest_output=xml:"$TEST_REPORTS_DIR"/"$case_name".xml 2>&1 | tee ${{ github.workspace }}/ut_log/binary_UT_${ut_suite}_${case_name}_Test.log
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:125]: add one more folder under `ut_log`, for example `ut_log/binary_ut/${case_name}...log`

File: .github/workflows/_linux_ut.yml [hunk 5]
```diff
@@ -142,3 +144,31 @@ jobs:
             test_cmd="${test_cmd} test_xpu.py"
           fi
           eval $test_cmd
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:146]: How about inductor test logs? will we upload those logs too? Seems the pytorch ut test log under `./test/test-reports/` by default

File: .github/workflows/_linux_ut.yml [hunk 6]
```diff
@@ -142,3 +144,31 @@ jobs:
...
+          }
+          set -xe
+          cp .github/scripts/ut_result_check.sh ${{ github.workspace }}/ut_log
+          for ut_suite in $(echo ${{ inputs.ut_suite }} |sed 's/,/ /g')
+          do
+            contains "op_example,op_extended,op_ut,torch_xpu" $ut_suite
+            $contains_status
+            cd ${{ github.workspace }}/ut_log
+            bash ut_result_check.sh ${ut_suite}
+          done
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:174]: Better to add a space line at the end of file

**Comment by chuanqi129** [comment]
build failure need rebase the code to the latest main

**Comment by fengyuan14** [comment]
Please wait for preci fixing.

**Comment by chuanqi129** [comment]
@RUIJIEZHONG66166 Can we transfer the status of this PR to ready for review? And can this PR address the issue https://github.com/intel/torch-xpu-ops/issues/1166?

**Pull request closed without merging**